src/imageio/storage/piwigo.c

```diff
@@ -706,8 +723,71 @@ static gboolean _piwigo_api_create_new_album(dt_storage_piwigo_params_t *p)
   return TRUE;
 }
 
+static char *_piwigo_api_get_image_id(dt_storage_piwigo_params_t *p, dt_image_t *img)
+{
+  GList *args = NULL;
+  char album_id[10];
+  sprintf(album_id, "%d", (int) p->album_id);
+
+  args = _piwigo_query_add_arguments(args, "method", "pwg.categories.getImages");
+  args = _piwigo_query_add_arguments(args, "cat_id", album_id);
+  args = _piwigo_query_add_arguments(args, "per_page", "999999");
+
+  _piwigo_api_post(p->api, args, NULL, TRUE);
+
+  g_list_free(args);
+
+  if(p->api->response && !p->api->error_occured && json_object_has_member(p->api->response, "result"))
+  {
+    JsonNode *result_node = json_object_get_member(p->api->response, "result");
+
+    if(result_node != NULL && json_node_get_node_type(result_node) == JSON_NODE_OBJECT)
+    {
+      JsonObject *result = json_node_get_object(result_node);
+
+      if(json_object_has_member(result, "images"))
+      {
+        JsonArray *existing_images = json_object_get_array_member(result, "images");
+
+        for(int i = 0; i < json_array_get_length(existing_images); i++)
+        {
+          JsonObject *existing_image = json_array_get_object_element(existing_images, i);
+          if(json_object_has_member(existing_image, "file"))
+          {
+            if(strcmp(img->filename, json_object_get_string_member(existing_image, "file")) == 0)
+            {
+              sprintf(existing_image_id, "%d", (int) json_object_get_int_member(existing_image, "id"));
```

Comment:
You can't use a global variable here. Please move declaration inside this routine.

Reply:
removed the variable